### ISSUE TITLE
Numpy docs fail to build with Sphinx 1.2

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1335]
When trying to build Numpy docs with the latest Sphinx 1.2, I get:

```
Exception occurred:
  File "/usr/lib/python2.7/dist-packages/jinja2/utils.py", line 389, in __getitem__
    if self._queue[-1] != key:
RuntimeError: maximum recursion depth exceeded in cmp
```

The full traceback mentioned that this happened while processing file `doc/source/_templates/autosummary/class.rst`.

Looks like this regression was caused by two commits: 97d5dd1 and b56a153 by [Takayuki Shimizukawa](https://bitbucket.org/shimizukawa). Reverting them fixes the issue.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1335
- Originally reported by: Dmitry Shachnev
- Originally created at: 2013-12-26T14:08:55.228

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-12-26 13:20:13+00:00_

OK, I'll get numpy docs and fix it.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-12-27 08:32:40+00:00_

Fix autosummary template overloading with exclamation prefix cause infinite recursive function call. Closes #1335

→ <<cset f58b7c43b9af9bd8ac1c8c04570575155f5c5dda>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-12-27 08:32:40+00:00_

Fix autosummary template overloading cause infinite recursive function call. Closes #1335

→ <<cset 3d67be3dcee3144edccf5d054e660a994f48bac9>>
